### ISSUE TITLE
Question: Is it possible to have on-demand reloading of skybox?

### ISSUE THREAD

**Issue by lawre** [KeenSoftwareHouse/SpaceEngineers#126]
I have yet to dig through the source code to find my answer, but just in-case anyone else has the same thought as me, here's my question:

Does anyone see that it might be possible for the server to periodically send a new skybox to all connected clients and have the client change to the newly uploaded skybox?  This would pave the way for some really interesting things:

1. Animated skyboxes... i.e, the nearest star or planet rotating.  Or the Star Trek "Warp drive" effect that uses distant stars and debris to show movement.  This would basically be a small movie in loop.

2. Scenario scenes... Show an approaching gigantic asteroid that gets bigger and bigger, you have to build an asteroid killer and launch it off... when the asteroid is destroyed the animation changes to that of destruction. This 

3. ??? Probably some other cool things that I don't have time on my lunch break to think of

**Comment by Spartan322** [comment]
What you are asking for is an image being loaded at runtime, which means the image (or animation, which is completely different, you should be using a gif-like filetype instead) is in memory, and could change only when some specific function is called (except an animation changes every specified time usually). This probably requires huge changes, especially changing the renderer. The other issue is if the skybox is changed on the server, it will not be handled (which a script would do, cause a script runs on the server) because SE doesn't handle client things as well (unless the skybox o the workshop is completely redesigned)

**Comment by midspace** [comment]
I looked trying this back in January after discovering that some Environmental stuff can be changed via the API, but the skybox wasn't part of the code that is refreshed when that property is changed.
There probably isn't any need to download new skyboxes, as *all* relevant skyboxes should be downloaded first from any mod or mods.
We just need to extend the API, to allow changing the selected skybox texture during runtime.
It is just a rendered background after all.
The expected Message classes should also be set up to sync between the server and clients for a global environment change.
It can be done.

**Comment by Spartan322** [comment]
@midspace I know it can be done, but it will take a while and you need to modify a lot (which if your not careful, will break backward compatibility, which always get closed), also lets not forget this feature could very easily drag the speed down, this feature has to implemented very either slowly or super carefully.

**Comment by joemorin73** [comment]
I have to agree with @Spartan322.  This feels like a feature that would be very heavy on the game.  This might be more of a "should you" vs "could you".

**Comment by KarryonDesigns** [comment]
This would be an amazing feature but like other said it would be pretty heavy on the game.  This may be possible after the game is optimized and allows for multi-core threading but until then It wouldn't even be something I would want to try.  If the game is optimized to use more than 4 cores this could be something that is very feasible to to.

**Comment by freekostia** [comment]
So much time passed. Now it seems to be possible to set custom skybox on demand by triggers from mod scripts. But there is no information how to use it yet. 
http://forum.keenswh.com/threads/setcustomskybox-function-how-to-use-it.7389260/
